### PR TITLE
Forward special P/Invoke attributes to the target DllImport method

### PR DESCRIPTION
**Pull request by jkoritzinsky**
If the GeneratedDllImport-attributed method has SuppressGCTransition or UnmanagedCallConv attributes, apply the attributes to the generated DllImport method.

### CODE REVIEW

**Comment by jkoritzinsky** [comment]
@AaronRobinsonMSFT looks like Clang warns about unused attributes (like calling conventions on x64), and since Clang uses a pretty standard warning format, MSBuild recognizes it and elevates it to an error since we have WarnAsError set to true.

Is there an easy way to suppress the warning from DNNE? Also, is it worth having integration tests for the calling convention work since we validate that we forwarded the attribute in the unit tests and the calling conventions would only have a difference in x86, which we don't run against?

**Comment by AaronRobinsonMSFT** [comment]
Updated DNNE to avoid setting these values - https://www.nuget.org/packages/DNNE/1.0.22.

File: DllImportGenerator/DllImportGenerator/DllImportGenerator.cs [hunk 1]
```diff
@@ -112,8 +130,11 @@ public void Execute(GeneratorExecutionContext context)
                     generatorDiagnostics.ReportConfigurationNotSupported(lcidConversionAttr, nameof(TypeNames.LCIDConversionAttribute));
                 }
 
+                // TODO: These need to be propagaged to the P/Invoke target, which is this does not do correctly.
```
Comment (AaronRobinsonMSFT) [inline DllImportGenerator/DllImportGenerator/DllImportGenerator.cs]: Why don't these get propagated properly? Spelling and grammar check please.

File: DllImportGenerator/DllImportGenerator/DllImportGenerator.cs [hunk 2]
```diff
@@ -127,6 +148,44 @@ public void Initialize(GeneratorInitializationContext context)
             context.RegisterForSyntaxNotifications(() => new SyntaxContextReceiver());
         }
 
+        private List<AttributeSyntax> GenerateSyntaxForForwardedAttributes(AttributeData? suppressGCTransitionAttribute, AttributeData? unmanagedCallConvAttribute)
+        {
+            const string CallConvsField = "CallConvs";
+            // Manually rehydrate the forwarded attributes with fully qualified types so we don't have to worry about any using directives.
+            List<AttributeSyntax> attributes = new List<AttributeSyntax>();
```
Comment (AaronRobinsonMSFT) [inline DllImportGenerator/DllImportGenerator/DllImportGenerator.cs]: `new()`?

**Review by AaronRobinsonMSFT — APPROVED** [review]

File: DllImportGenerator/DllImportGenerator/DllImportGenerator.cs (hunk 1 above)
Comment (jkoritzinsky) [reply DllImportGenerator/DllImportGenerator/DllImportGenerator.cs]: This is an old todo that I need to remove (I've fixed this)

**Pull request merged**